packages/dart/lib/src/protocol/simple_span.dart

```diff
@@ -1,29 +1,67 @@
 import '../../sentry.dart';
 
 class SimpleSpan implements Span {
+  final SpanId _spanId;
+  final Hub _hub;
   @override
-  void end({DateTime? endTimestamp}) {
-    // TODO: implement end
-  }
+  final Span? parentSpan;
+  final Map<String, SentryAttribute> _attributes = {};
+
+  String _name;
+  SpanV2Status _status = SpanV2Status.ok;
+  DateTime? _endTimestamp;
+  bool _isFinished = false;
+
+  SimpleSpan({
+    required String name,
+    this.parentSpan,
+    Hub? hub,
+  })  : _spanId = SpanId.newId(),
+        _hub = hub ?? HubAdapter(),
+        _name = name;
 
   @override
-  void setAttribute(String key, SentryAttribute value) {
-    // TODO: implement setAttribute
-  }
+  SpanId get spanId => _spanId;
 
   @override
-  void setAttributes(Map<String, SentryAttribute> attributes) {
-    // TODO: implement setAttributes
+  String get name => _name;
+
+  @override
+  set name(String value) => _name = value;
+
+  @override
+  SpanV2Status get status => _status;
+
+  @override
+  set status(SpanV2Status value) => _status = value;
+
+  @override
+  DateTime? get endTimestamp => _endTimestamp;
+
+  @override
+  Map<String, SentryAttribute> get attributes => Map.unmodifiable(_attributes);
+
+  @override
+  bool get isFinished => _isFinished;
+
+  @override
+  void setAttribute(String key, SentryAttribute value) {
+    _attributes[key] = value;
   }
 
   @override
-  void setName(String name) {
-    // TODO: implement setName
+  void setAttributes(Map<String, SentryAttribute> attributes) {
```

Comment:
Will there also be `removeAttribute/removeAttributes`?  Currently, there is no way to remove attributes.